File: dotnet/Entites/ItineraryItem.cs
namespace dotnet.Entites
{
    public class ItineraryItem : BaseAuditableEntity
    {
        public Guid Id { get; set; }
        public Guid ItineraryDayId { get; set; }
        public string? PlaceId { get; set; }
        public TimeOnly? StartTime { get; set; }
        // Duration to handle cross days activity
        public TimeSpan? Duration { get; set; }

        // AI-generated or user-added note (e.g. "Check-in & rest", "Explore Old Quarter")
        public string? Note { get; set; }

        // Google Calendar sync
        public string? GoogleCalendarEventId { get; set; }
        public bool IsDeleted { get; set; } = false;

        public ItineraryDay ItineraryDay { get; set; }
    }
}

File: react/types/itineraryItem.ts
import { Place } from "@/types/place";

export type ItineraryItem = {
  id: string;
  itineraryDayId: string;
  place?: Place | null;
  startTime?: string; // TimeOnly HH:mm (optional)
  duration?: string; // TimeOnly HH:mm (optional)
  note?: string | null; // AI-generated or user note
  googleCalendarEventId?: string;
};
